### PR TITLE
feat(train_test_split): Add "shuffle is False" warning

### PR DESCRIPTION
**Pull request by augustebaum**
Addresses #684

### CODE REVIEW

**Comment by augustebaum** [comment]
Closed by https://github.com/probabl-ai/skore/issues/492#issuecomment-2491444726
